controllers/bsl.go

```diff
@@ -33,34 +34,46 @@ func (r *DPAReconciler) ValidateBackupStorageLocations(log logr.Logger) (bool, e
 	// Ensure BSL is a valid configuration
 	// First, check for provider and then call functions based on the cloud provider for each backupstoragelocation configured
 	for _, bslSpec := range dpa.Spec.BackupLocations {
-		provider := bslSpec.Velero.Provider
-		if len(provider) == 0 {
-			return false, fmt.Errorf("no provider specified for one of the backupstoragelocations configured")
-		}
+		if bslSpec.Velero != nil {
+			provider := bslSpec.Velero.Provider
+			if len(provider) == 0 {
+				return false, fmt.Errorf("no provider specified for one of the backupstoragelocations configured")
+			}
 
-		// TODO: cases might need some updates for IBM/Minio/noobaa
-		switch provider {
-		case AWSProvider, "velero.io/aws":
-			err := r.validateAWSBackupStorageLocation(*bslSpec.Velero, &dpa)
-			if err != nil {
-				return false, err
+			// TODO: cases might need some updates for IBM/Minio/noobaa
+			switch provider {
+			case AWSProvider, "velero.io/aws":
+				err := r.validateAWSBackupStorageLocation(*bslSpec.Velero, &dpa)
+				if err != nil {
+					return false, err
+				}
+			case AzureProvider, "velero.io/azure":
+				err := r.validateAzureBackupStorageLocation(*bslSpec.Velero, &dpa)
+				if err != nil {
+					return false, err
+				}
+			case GCPProvider, "velero.io/gcp":
+				err := r.validateGCPBackupStorageLocation(*bslSpec.Velero, &dpa)
+				if err != nil {
+					return false, err
+				}
+			default:
+				return false, fmt.Errorf("invalid provider")
 			}
-		case AzureProvider, "velero.io/azure":
-			err := r.validateAzureBackupStorageLocation(*bslSpec.Velero, &dpa)
-			if err != nil {
-				return false, err
+		}
+		if bslSpec.Bucket != nil {
+			// Make sure credentials are specified.
+			if bslSpec.Bucket.Credential == nil {
+				return false, fmt.Errorf("must provide a valid credential secret")
 			}
-		case GCPProvider, "velero.io/gcp":
-			err := r.validateGCPBackupStorageLocation(*bslSpec.Velero, &dpa)
-			if err != nil {
-				return false, err
+			if bslSpec.Bucket.Credential.LocalObjectReference.Name == "" {
+				return false, fmt.Errorf("must provide a valid credential secret name")
 			}
-		// case when no supporting cloud provider is configured in bsl
-		default:
-			return false, fmt.Errorf("no valid provider configured for one of the backupstoragelocations")
+		}
+		if bslSpec.Bucket != nil && bslSpec.Velero != nil {
```

Comment:
is there ever a case where both spec.Bucket and spec.Velero are nil that we want to handle here?

Reply:
I believe that this is covered in the ensureMapping function here: https://github.com/openshift/oadp-operator/pull/432/files#diff-92d442dfa6f9e391ba11832929f440817bcabe3f794700a7e9ecd9a10c309d24R293-R294